aiosmtpd/tests/test_handlers.py

```diff
@@ -186,6 +188,65 @@ def test_message_decoded(self):
         self.assertEqual(self.handled_message['X-RcptTos'], 'bart@example.com')
 
 
+class TestAsyncMessage(unittest.TestCase):
+    def setUp(self):
+        self.handled_message = None
+
+        class MessageHandler(AsyncMessage):
+            @asyncio.coroutine
+            def handle_message(handler_self, message, loop):
+                self.handled_message = message
+
+        self.handler = MessageHandler()
```

Comment:
I'm not a big fan of vertical whitespace for separation.  I'll suggest adding comments instead, or restructuring the test to make the subclass a module global, or just collapsing out the whitespace (it's not generally too hard to see the difference).  But I'm open to thoughts here.

Also, It seems like you could create the controller, start it, and arrange for it to be stopped in an `addCleanup()` all in the `setUp()` method.  That would reduce some duplication in the test methods.